utils/build.ps1

```diff
@@ -254,16 +254,47 @@ function Invoke-BuildStep([string]$Name) {
   }
 }
 
-enum Target {
+enum TargetComponent {
   LLVM
   Runtime
   Dispatch
   Foundation
   XCTest
 }
 
-function Get-ProjectBinaryCache($Arch, [Target]$Target) {
-  return "$BinaryCache\" + ($Arch.BuildID + $Target.value__)
+function Get-TargetProjectBinaryCache($Arch, [TargetComponent]$Project) {
+  return "$BinaryCache\" + ($Arch.BuildID + $Project.value__)
+}
+
+enum HostComponent {
+  BuildTools
+  Compilers
+  System
+  ToolsSupportCore
+  LLBuild
+  Yams
+  ArgumentParser
+  Driver
+  Crypto
+  Collections
+  ASN1
+  Certificates
+  PackageManager
+  Markdown
+  Format
+  IndexStoreDB
+  SourceKitLSP
+  LMDB
+  SymbolKit
+  DocC
+}
+
+function Get-HostProjectBinaryCache([HostComponent]$Project) {
+  return "$BinaryCache\$($Project.value__)"
+}
+
+function Get-HostProjectCMakeModules([HostComponent]$Project) {
+  return "$BinaryCache\$($Project.value__)\cmake\modules"
```

Comment:
All remaining instances of hard-coded values in this file are to `$BinaryCache\1` or `$BinaryCache\7`, so how about defining two globals after this function definition:

```ps1
$CompilersBinaryCache = Get-HostProjectBinaryCache Compilers
$DriverBinaryCache = Get-HostProjectBinaryCache Driver

Reply:
I'm not sure if the usages are in the same function, I was suggesting creating global variables for them. (but even better if they can be locals)